src/Chapter2.hs

```diff
@@ -837,12 +876,18 @@ list.
 
 >>> rotate 1 [1,2,3,4]
 [2,3,4,1]
+
 >>> rotate 3 [1,2,3,4]
 [4,1,2,3]
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+
+rotate :: Int -> [a] -> [a]
+rotate n xs
+    | n < 0 = []
+    | null xs = []
+    | otherwise = take (length xs) $ drop n $ cycle xs
```

Comment:
Great! 
In order to make it even more efficient, you can check another corner case: if `n` is equal to 0 (by mod length).
Also, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
Hmm ...,

How about this?

```haskell
rotate n xs
    | n < 0 = []
    | null xs = []
    | otherwise = take l $ drop (n `mod` l) $ cycle xs
    where
      l = length xs
```

Thank you.

Reply:
Yeap, this is awesome ⭐